upass/runner/upass_runner_cycle_test.cpp

```
@@ -90,3 +90,151 @@ TEST(UpassRunnerResolve, SharedDecideResolves) {
+// `cond_text` is the condition literal ("true", "false", or a ref name like "___c").
+// `else_val`  is "" when there is no else branch.
+static std::shared_ptr<Lnast> make_if_lnast(std::string_view cond_text,
+                                             std::string_view then_val  = "4",
+                                             std::string_view else_val  = "") {
+  auto ln = std::make_shared<Lnast>("if_test");
+  ln->set_root(Lnast_node::create_top());
+  auto stmts = ln->add_child(Lnast_nid::root(), Lnast_node::create_stmts());
+  auto if_nid = ln->add_child(stmts, Lnast_node::create_if());
+
+  // Condition — emit as const ("true"/"false") or ref (unknown variable).
+  bool is_literal = (cond_text == "true" || cond_text == "false");
```

Cppcheck (2.20.0)

[error] Non-local variable 'fname' will use object that points to local variable '_fname'.

(danglingLifetime)